routes/institutions/institutionsUtility.js

```diff
@@ -8,18 +8,19 @@ const handleFieldAlreadyInUseErrorInfo = (err, res) => {
 	else if (fieldAlreadyInUse === 'domain') repeatedFieldErrorCode = 'E1203';
 	else if (fieldAlreadyInUse === 'secondaryDomain') repeatedFieldErrorCode = 'E1204';
 	else return res.status(500).send({ error: errorCodes['E1206'] });
+				
 	return res.status(400).send({ error: errorCodes[repeatedFieldErrorCode] });
 };
 
 const validateInstitutionFields = (institutionName, domain, secondaryDomain) => {
 	const isMandatoryFieldsValid = typeof institutionName === 'string' && typeof domain === 'string';
-	const isSecondaryDomainValid = !secondaryDomain || typeof secondaryDomain === 'string';
+	const isSecondaryDomainValid = secondaryDomain && typeof secondaryDomain === 'string';
```

Comment:
Im guessing that you guys have a place where secondary domain is required for some reason. If that is the case let me know and we will have to keep the validation as open as possible and do the check elsewhere

Reply:
**Yes** the Secondary Domain should be also be valid when its empty or just " ". 
The secondary domain should never be required anywere, but only as optional. And If it fails, it's because someone coded wrongly, and it shouldnt be Allowed to be pushed. 
It should be very clear for everyone that Secondary Domain is Optional XD

So please fix that.

Reply:
See comment: https://github.com/Educado-App/educado-backend/pull/189#issuecomment-2474100970